### DISCUSSION TITLE
用户管理是否支持过滤 Microsoft Active Directory 禁用用户 ？

### DISCUSSION THREAD

**Discussion by ganzaofang** [TencentBlueKing/bk-user#184] [Q&A] [answered]
**用文字描述你遇到的问题**
请问下配置Microsoft Active Directory对接时，能否在用户对象过滤那把禁用的AD用户过滤掉呢？我根据原格式增加Enabled的字段(&(objectCategory=Person)(sAMAccountName=*)(Enabled=True)) ，但测试不生效。请问是否没有这个功能，谢谢！

**版本**
bk_user_manage  2.2.6b3
bk-usermgr          2.2.6b3

**原贴地址**
https://bk.tencent.com/s-mart/community/question/5044?type=answer

**Comment by Canway-shiisa** [comment, marked as answer]
1. 用户管理-AD目录同步的用户对象配置是支持过滤的，过滤使用ad原生语法；
2. 要进行过滤禁用用户后同步，可以参考下面的配置，在用户目录-字段配置-用户字段类 增加 `(!(userAccountControl=514))`

![image](https://user-images.githubusercontent.com/90179140/144352772-40917f34-ff3e-4c58-acaa-5af77466dac9.png)

> 另：不是所有的AD系统 禁用用户的属性值 (userAccountControl) 都是 `514`，存在某些 AD 系统使用 `60082` 代表禁用值，`60080` 代表启用值；因此，具体实施还要根据使用的AD系统来进行配置
